src/common.h

```diff
@@ -162,10 +158,16 @@ struct Location {
       size_t offset;
     };
   };
+
+
+  Location() {
+    this->line = 0;
```

Comment:
Turns out I can initialize one union or the other, but not both if I use initializers so I'm doing the larger union. Another option is to have no constructor and "inline" the initialization of filename with:
```
const char* filename = NULL;
```
That works and passes all tests. Maybe you can ask a C++ Guru if that also guarantees zero initialization of the union. From [the first answer here](https://stackoverflow.com/questions/4203010/how-to-initialize-member-struct-in-initializer-list-of-c-class) I think it might.